Libraries/Image/AssetSourceResolver.js

```diff
@@ -123,6 +134,10 @@ class AssetSourceResolver {
     );
   }
 
+  assetFromCatalog(): ResolvedAssetSource {
```

Comment:
It's confusing this is using `getAndroidResourceIdentifier` in an iOS code path. If this is right, can we rename `getAndroidResourceIdentifier` to be platform-independent?

```suggestion
  assetFromIOSCatalog(): ResolvedAssetSource {
```

Reply:
I renamed it in the initial PR, but it added a lot of complexity to land this since it is in a separate package so we decided to leave it as is for now and land the name change in a follow up.

Reply:
Could you rename the method to `getResourceIndentifier` and a `getAndroidResourceIdentifier` that just calls the original method?